usaspending_api/download/v2/list_unlinked_awards_downloads.py

```diff
@@ -0,0 +1,106 @@
+import re
+from typing import Optional
+import boto3
+
+from django.conf import settings
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.config import CONFIG
+from usaspending_api.download.filestreaming.s3_handler import S3Handler
+from usaspending_api.references.models import ToptierAgency
+
+
+class ListUnlinkedAwardsDownloadsViewSet(APIView):
+    """
+    Returns a list which contains links to the latest versions of unlinked awards files for an agency.
+    """
+
+    s3_handler = S3Handler(
+        bucket_name=settings.BULK_DOWNLOAD_S3_BUCKET_NAME, redirect_dir=settings.UNLINKED_AWARDS_DOWNLOAD_REDIRECT_DIR
+    )
+
+    # This is intentionally not cached so that the latest updates to these files are always returned
+    def post(self, request):
+        """Return list of downloads that match the requested params."""
+        toptier_code = request.data.get("toptier_code", None)
+
+        # Check required params
+        required_params = {"toptier_code": toptier_code}
+        for required, param_value in required_params.items():
+            if param_value is None:
+                raise InvalidParameterException(f"Missing one or more required body parameters: {required}")
+
+        agency_check = ToptierAgency.objects.filter(toptier_code=toptier_code).values(
+            "toptier_code", "name", "abbreviation"
+        )
+        if not agency_check or len(agency_check.all()) == 0:
+            raise InvalidParameterException(f"Agency not found given the toptier code: {toptier_code}")
+        agency = agency_check[0]
+
+        # Populate regex
+        download_prefix = f"{agency['name'].replace(' ', '_')}_unlinked_awards"
```

Comment:
This prefix is a little different than what is suggested in DEV-10589's description `Department_of_Veterans_Affairs_UnlinkedAwards_2024-01-29_H17M24S53136993.csv`. 

I think the format in the ticket is a little more consistent with out other download naming conventions (Pascal Case on the type of Download).

Reply:
This format is what the wanted though. This format came out of the call we all had together.

Reply:
Done